### PR TITLE
fix(git): Don't quote unicode paths

### LINKED ISSUE #15538: Japanese file names are renamed incorrectly every time in PR.
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Japanese file names are renamed incorrectly every time in pull requests.
I show an example below.

original file name:
`技術選定.md`

first PR commit diff on console:

```
rename from 技術選定.md
rename to "\"\\346\\212\\200\\350\\241\\223\\351\\201\\270\\345\\256\\232.md\""
```

second PR commit diff on console:

```
rename from "\"\\346\\212\\200\\350\\241\\223\\351\\201\\270\\345\\256\\232.md\""
rename to "\"\\\"\\\\346\\\\212\\\\200\\\\350\\\\241\\\\223\\\\351\\\\201\\\\270\\\\345\\\\256\\\\232.md\\\"\""
```

and more...

diff image on PR page:

スクリーンショット 2022-05-11 11 23 27

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### LINKED ISSUE #18805: File with umlaut ü (special character) in its name gets corruped
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### If you're self-hosting Renovate, select which platform you are using.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

My repo contains a file with an umlaut `ü` in its name. In the PRs that renovate creates, the files get corrupted.

grafik

Example: https://github.com/cypressious/RenovateUmlautIssue/pull/3

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

See [core.quotePath](https://git-scm.com/docs/git-config#Documentation/git-config.txt-corequotePath) Git option.

## Context

- Fixes: #18805
- fixes https://github.com/renovatebot/renovate/issues/15538

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Example: https://github.com/renovate-testing/test-18805-RenovateUmlautIssue/pulls

File: lib/util/git/index.ts [hunk 1]
```diff
@@ -237,8 +237,8 @@ export async function initRepo(args: StorageConfig): Promise<void> {
   const { localDir } = GlobalConfig.get();
   git = simpleGit(localDir, simpleGitConfig()).env({
     ...process.env,
-    LANG: 'C',
-    LC_ALL: 'C',
+    LANG: 'C.UTF-8',
+    LC_ALL: 'C.UTF-8',
```
Comment (zharinov) [inline lib/util/git/index.ts:241]: Not necessary, but seems like we may want it

**Comment by rarkins** [comment]
Any potential problems? Eg package files with spaces on their name stop working or something like that

**Comment by zharinov** [comment]
Don't think so. Everything that follows SHA is treated as file name by the regex, including spaces

**Review by rarkins — DISMISSED** [review]

**Review by JamieMagee — DISMISSED** [review]

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**